tests/sdk/test_add.py

```diff
@@ -110,10 +109,28 @@ def test_add_data_sample_already_exists(client, data_sample_query, mocker):
     assert m.is_called()
 
 
+def test_add_data_sample_with_paths(client, data_samples_query, mocker):
+    server_response = [{"key": "42"}]
+    m = mock_requests(mocker, "post", response=server_response)
+    with pytest.raises(ValueError):
+        client.add_data_sample(data_samples_query)
+
+    assert m.is_called()
```

Comment:
It took me a while to understand what this was doing. Could you add a comment explaining that you're intentionally passing `data_samples_query` instead of `data_sample_query` in order to check that you get an exception?

Also, I don't think `m` should be called at all. If it is then something went wrong in the validation.

Reply:
I'll add comments, it will indeed be clearer.
But I'm not sure about the second part of your comment, should I just remove the variable (and the related assert) or only be sure that it is properly called and checked?

Reply:
According to the code of  `client.add_data_sample`, the validation raising the ValueError occurs before any call is made to the API. Which means that the mocker should effectively never be called at all.

I tried this test:

```python
def test_add_data_sample_with_paths(client, data_samples_query, mocker):
    server_response = [{"key": "42"}]
    m = mock_requests(mocker, "post", response=server_response)
    assert m.is_called()
```

And it passes ! Even though there is absolutely no call!
This is a big issue, I'll create a new issue for it.

That being said, in this very case, since we're expecting the validation to break the call to add_data_sample, we don't need a mock response at all, the test could be simplified as:
```python
def test_add_data_sample_with_paths(client, data_samples_query, mocker):
    with pytest.raises(ValueError):
        client.add_data_sample(data_samples_query)
```
And all tests also expecting the validation to raise an exception could be simplified in a similar way.

Reply:
Indeed, I'll remove the `m.is_called()` and the `mock_requests` for these two tests!
I'll have a look to the others test that raises exceptions as well 🙂